src/DataMapper/ActivityMapper.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GroupLife\Core\DataMapper;
+
+use GroupLife\Core\Activity;
+
+class ActivityMapper
+{
+    private $connection;
+    private $scheduleMapper;
+    private $leaderMapper;
+
+    public function __construct(\Doctrine\DBAL\Connection $connection)
+    {
+        $this->connection = $connection;
+        $this->scheduleMapper = new ScheduleMapper($connection);
+        $this->leaderMapper = new LeaderMapper($connection);
```

Comment:
You delegate part of the functionality to `ScheduleMapper` and `LeaderMapper`, it's good. But better to inverse the dependency: instead of instantiating those mappers in ActivityMapper's constructor just pass them as an argument. This way you'll create loose coupling and make possible to unit-test this object.

```
public function __construct(\Doctrine\DBAL\Connection $connection, ScheduleMapper, LeaderMapper)
{}
```